### PR TITLE
Make `BucketThrottle` generic and introduce `ExpiryThrottle`

### LINKED ISSUE #3740: Generalize `ThrottleBucket` to work with operation types other than `HederaFunctionality`
### Problem

To generalize our deterministic throttles to apply to work done in auto-renew/expiration, we need to support arbitrary sets of operations, not only HAPI operation types.

### Solution

Generic-ize `ThrottleBucket` to work with any `enum` that defines a set of operations.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Makes `BucketThrottle` and `ThrottleGroup` generic so a [new expiry throttle](https://github.com/hashgraph/hedera-services/blob/03740-genericize-throttling/hedera-node/src/main/resources/expiry-throttle.json) can be used for operations of [type `MapAccessType`](https://github.com/hashgraph/hedera-services/blob/03740-genericize-throttling/hedera-node/src/main/java/com/hedera/services/throttling/MapAccessType.java#L26).
     * The new `expiry.throttleResource=expiry-throttle.json` property controls the throttle specification, and by default loads the above JSON.
     * If overridden to e.g. `expiry.throttleResource=0.0.1234`, the specification will come from UTF-8 encoded JSON in file `0.0.1234`.
 - Exposes the `MapAccessType` throttle via the [`ExpiryThrottle` singleton](https://github.com/hashgraph/hedera-services/blob/03740-genericize-throttling/hedera-node/src/main/java/com/hedera/services/throttling/ExpiryThrottle.java#L38) whose usage snapshot is [synced to the mutable `MerkleNetworkContext`](https://github.com/hashgraph/hedera-services/blob/03740-genericize-throttling/hedera-node/src/main/java/com/hedera/services/state/expiry/EntityAutoRenewal.java#L73) at the end of every `handleTransaction`.
 - Saves the expiry throttle's usage snapshot in the `MerkleNetworkContext` to allow deterministic expiry decisions to be made across reconnects.

**Related issues:**
 - Closes #3740

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java [hunk 1]
```diff
@@ -260,6 +277,7 @@ public void discardPreparedUpgradeMeta() {
         preparedUpdateFileHash = NO_PREPARED_UPDATE_FILE_HASH;
     }
 
+    @SuppressWarnings("java:S2696")
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java:280]: Is this because of sonar ?

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

File: hapi-utils/src/main/java/com/hedera/services/sysfiles/domain/throttling/ThrottleGroup.java [hunk 2]
```diff
@@ -15,14 +15,22 @@
...
     int opsPerSec;
     long milliOpsPerSec;
-    List<HederaFunctionality> operations = new ArrayList<>();
+    List<E> operations = new ArrayList<>();
+
+    public ThrottleGroup() {
+        // Needed by Jackson
+    }
+
+    public ThrottleGroup(long milliOpsPerSec, List<E> operations) {
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hapi-utils/src/main/java/com/hedera/services/sysfiles/domain/throttling/ThrottleGroup.java:30]: Addressed in https://github.com/hashgraph/hedera-services/pull/3842 now

File: hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java:280]: Yep!

**Pull request merged**